tests_config/run_spec.js

```diff
@@ -300,15 +291,46 @@ function parse(source, options) {
   return prettier.__debug.parse(source, options, /* massage */ true).ast;
 }
 
-function format(source, filename, options) {
-  const result = prettier.formatWithCursor(source, {
-    filepath: filename,
+function format(text, options) {
+  options = {
     ...options,
+  };
+
+  const inputWithCursor = text
+    .replace(RANGE_START_PLACEHOLDER, (match, offset) => {
+      options.rangeStart = offset;
+      return "";
+    })
+    .replace(RANGE_END_PLACEHOLDER, (match, offset) => {
+      options.rangeEnd = offset;
+      return "";
+    });
+
+  const input = inputWithCursor.replace(CURSOR_PLACEHOLDER, (match, offset) => {
+    options.cursorOffset = offset;
+    return "";
   });
```

Comment:
Actually this part aoild not right, if cursor is before `rangeEnd`, it is wrong.